From cb3f562246da5830a58f07e00af437034fcaba94 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:28 -0700
Subject: [PATCH 1653/1852] v4l: xilinx: rgb2yuv: Fix warnings for NULL_RETURNS

commit 00a09e3b3cd690b39e1e2270f5cde4ee7ec6ed69 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings are:

For __xrgb2yuv_get_pad_format():
Event return_null: 	Explicitly returning null.

For xrgb2yuv_set_format():
Event returned_null: 	"__xrgb2yuv_get_pad_format" returns "NULL"
Event var_assigned: 	Assigning: "format" = "NULL" return value from
"__xrgb2yuv_get_pad_format".
Event dereference: 	Dereferencing "format", which is known to be
"NULL"

For xrgb2yuv_get_format():
Event returned_null: 	"__xrgb2yuv_get_pad_format" returns "NULL".
Event dereference: 	Dereferencing
"__xrgb2yuv_get_pad_format(xrgb2yuv, cfg, fmt->pad, fmt->which)", which
is known to be "NULL"

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-rgb2yuv.c    | 26 ++++++++++++++-----
 1 file changed, 20 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-rgb2yuv.c b/drivers/media/platform/xilinx/xilinx-rgb2yuv.c
index 20ae95946ca3..ebcfb497bf5d 100644
--- a/drivers/media/platform/xilinx/xilinx-rgb2yuv.c
+++ b/drivers/media/platform/xilinx/xilinx-rgb2yuv.c
@@ -98,15 +98,22 @@ __xrgb2yuv_get_pad_format(struct xrgb2yuv_device *xrgb2yuv,
 			  struct v4l2_subdev_pad_config *cfg,
 			  unsigned int pad, u32 which)
 {
+	struct v4l2_mbus_framefmt *format;
+
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_format(&xrgb2yuv->xvip.subdev, cfg,
-						  pad);
+		format = v4l2_subdev_get_try_format(&xrgb2yuv->xvip.subdev,
+						    cfg, pad);
+		break;
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
-		return &xrgb2yuv->formats[pad];
+		format = &xrgb2yuv->formats[pad];
+		break;
 	default:
-		return NULL;
+		format = NULL;
+		break;
 	}
+
+	return format;
 }
 
 static int xrgb2yuv_get_format(struct v4l2_subdev *subdev,
@@ -114,9 +121,14 @@ static int xrgb2yuv_get_format(struct v4l2_subdev *subdev,
 			       struct v4l2_subdev_format *fmt)
 {
 	struct xrgb2yuv_device *xrgb2yuv = to_rgb2yuv(subdev);
+	struct v4l2_mbus_framefmt *format;
+
+	format = __xrgb2yuv_get_pad_format(xrgb2yuv, cfg, fmt->pad,
+					   fmt->which);
+	if (!format)
+		return -EINVAL;
 
-	fmt->format = *__xrgb2yuv_get_pad_format(xrgb2yuv, cfg, fmt->pad,
-						 fmt->which);
+	fmt->format = *format;
 
 	return 0;
 }
@@ -129,6 +141,8 @@ static int xrgb2yuv_set_format(struct v4l2_subdev *subdev,
 	struct v4l2_mbus_framefmt *format;
 
 	format = __xrgb2yuv_get_pad_format(xrgb2yuv, cfg, fmt->pad, fmt->which);
+	if (!format)
+		return -EINVAL;
 
 	if (fmt->pad == XVIP_PAD_SOURCE) {
 		fmt->format = *format;
-- 
2.31.1

